isis/tests/FunctionalTestsReduce.cpp

```diff
@@ -0,0 +1,217 @@
+#include "Fixtures.h"
+#include "Pvl.h"
+#include "PvlGroup.h"
+#include "TestUtilities.h"
+#include "Histogram.h"
+
+#include "reduce_app.h"
+
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+
+using namespace Isis;
+using ::testing::HasSubstr;
+
+static QString APP_XML = FileName("$ISISROOT/bin/xml/reduce.xml").expanded();
+
+TEST_F(LargeCube, FunctionalTestReduceDefault) {
+  QTemporaryDir prefix;
+  QString outCubeFileName = prefix.path() + "/outTemp.cub";
+  QVector<QString> args = {"from=" + testCube->fileName(),
+                            "to=" + outCubeFileName,
+                            "algorithm=average",
+                            "mode=total",
+                            "ons=100",
+                            "onl=100"
+                          };
+
+  UserInterface options(APP_XML, args);
+  Pvl log;
```

Comment:
Since we don't use the log in the tests, can we remove the log variables?

Reply:
I believe the reason we have log variables in the tests is due to needing to run reduce_app which requires a log.

Reply:
Because log defaults to a nullptr, if you don't pass one in, it should set to null.

Reply:
An alternative would be to specify a default value of nullptr in the header and remove the log variables in the tests. Let me know if you had something else in mind.

Reply:
+1 That's exactly what I was thinking.